theano/compile/function_module.py

```diff
@@ -133,6 +133,11 @@ def __init__(self, protected):
         self.protected = list(protected)
 
     def validate(self, fgraph):
+        if config.cycle_detection == 'fast' and hasattr(fgraph, 'has_destroyers'):
+            if fgraph.has_destroyers(self.protected):
+                raise gof.InconsistencyError("Trying to destroy a protected"
+                                             "Variable.")
+            return True
         if not hasattr(fgraph, 'destroyers'):
             return True
         for r in self.protected + list(fgraph.outputs):
```

Comment:
With the other change proposed, we could replace the full implementation of this method to do the same thing for all values of cycle_detection.